studies/models.py

```diff
@@ -72,6 +76,32 @@ class Meta:
             ('can_view_demographics', 'Can View Demographics'),
         )
 
+    @property
+    def begin_date(self):
+        active_logs = self.logs.filter(action='active')
+        if active_logs:
+            return active_logs.latest('action').created_at
+        else:
+            return None
+
+    @property
+    def end_date(self):
+        begin_date = self.begin_date
+        deactivated_logs = self.logs.filter(action='deactivated')
+        if deactivated_logs and begin_date:
+            end_date = deactivated_logs.latest('action').created_at
+        else:
+            return None;
+        return end_date if end_date > begin_date else None
+
+    @property
+    def completed_responses(self):
```

Comment:
I might rename this to `completed_responses_count` or `num_completed_responses` so it doesn't sound like it's going to actually return the responses. Same with the one below.